### PR TITLE
refactor: rename settings (breaking change)

### PR DESCRIPTION
**Pull request by hacdias**
Related to https://github.com/filebrowser/filebrowser/pull/485#discussion_r207921156, filebrowser/filebrowser.github.io#9 and filebrowser/caddy#4

License: MIT
Signed-off-by: Henrique Dias <hacdias@gmail.com>

### CODE REVIEW

**Comment by eine** [comment]
This PR modifies `caddy/parser/parser.go`, which is removed in #492.

- If #492 is merged first, these changes need to be done in `filebrowser/caddy` directly.
- If this PR is merged first, changes need to be done in `filebrowser/caddy` as soon (or before) #492 is merged.

**Comment by hacdias** [comment]
Wait for #492

**Comment by eine** [comment]
A thought before actually reviewing it: since this is a breaking change, and the filebrowser docker image is automatically updated now, should we publish a release (`v.1.10.0) before merging this? That would provide users, who cannot update their configs ATM, a stable image to use.

NOTE: when we publish a release now, a single docker image is created and it is tagged twice (`latest` and `$version`). So, if we publish a release now and a breaking one in a few days, we will have one tagged `v.1.10.0` and another one tagged `v2.0.0` and `latest`.

Furthermore, I don't know if the changes introduced in #485 should be considered breaking changes too.

**Review by eine — APPROVED** [review]

**Comment by hacdias** [comment]
Yes @1138-4EB, please do the honors 😄

**Comment by Equim-chan** [comment]
Might need a db mitigation, since the structure has been changed?

**Comment by hacdias** [comment]
@Equim-chan no, the DB wasn't changed at all 😄 just the CLI options and the config file

**Review by Equim-chan — APPROVED** [review]

File: cmd/filebrowser/main.go [hunk 1]
```diff
@@ -53,70 +56,82 @@ func init() {
...
+	flag.StringVar(&locale, "defaults.locale", "", "Default locale for new users, set it empty to enable auto detect from browser")
+
+	// Recaptcha settings
+	flag.StringVar(&recaptcha.host, "recaptcha.host", "https://www.google.com", "Use another host for ReCAPTCHA. recaptcha.net might be useful in China")
+	flag.StringVar(&recaptcha.key, "recaptcha.key", "", "ReCaptcha site key")
+	flag.StringVar(&recaptcha.secret, "recaptcha.secret", "", "ReCaptcha secret")
+
+	// Auth settings
+	flag.StringVar(&auth.method, "auth.method", "default", "Switch between 'none', 'default' and 'proxy' authentication")
+	flag.StringVar(&auth.header, "auth.header", "X-Forwarded-User", "The header name used for proxy authentication")
```
Comment (Equim-chan) [inline cmd/filebrowser/main.go:81]: Can we make some deprecated flags? For example when user tries to use the old `--allow-new` flag, we give a hint that `--allow-new is deprecated, use --defaults.allowNew instead`.

Comment (hacdias) [reply cmd/filebrowser/main.go:81]: I think we could, but we'd need to do it for every flag. If you're at it, you can update this PR itself 😄

**Comment by eine** [comment]
@hacdias, can you please have a look at the comment about [BindPFlags](https://godoc.org/github.com/spf13/viper#BindPFlags) in #506?

Also, I don't know if you prefer to merge this PR on its own, or as part of #506, since it is almost ready too.

**Comment by hacdias** [comment]
@1138-4EB yes, it's ok for me to close this in favor of #506 😄 and using BindPFlags as long as it works

File: cmd/filebrowser/main.go (hunk 1 above)
Comment (eine) [reply cmd/filebrowser/main.go:81]: I am at it. Will push soon.

https://godoc.org/github.com/spf13/pflag#FlagSet.MarkDeprecated

Comment (eine) [reply cmd/filebrowser/main.go:81]: Done. See https://github.com/filebrowser/filebrowser/pull/506/commits/1314e432be7235af20c35b5d815b7b4a73cda059.

**Pull request closed without merging**